platform/mds/mds/src/main/java/org/motechproject/mds/service/MotechDataService.java

```diff
@@ -218,14 +218,20 @@
     Class<T> getClassType();
 
     /**
-     * Evicts all cached entities
-     *
+     * Evicts all cached entities. This affects all entities.
      */
-    void evictAll();
+    void evictAllCache();
 
     /**
-     * Evicts the provided entity from the cache
+     * Evicts cache for a single entity instance.
      *
+     * @param instance the instance to clear the cache for
+     */
+    void evictCacheForInstance(T instance);
+
+    /**
+     * Evicts cache for the entity class of this data service.
+     * @param withSubclasses if true, the cache for subclasses of the entity will be also cleared
      */
-    void evictOne(T object);
+    void evictEntityCache(boolean withSubclasses);
```

Comment:
not sure I like hiding the feature to evict sub classes inside an argument for the method call and would rather have it be 2 calls... but then again, it didn't kill anyone.